src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/CreateTask.groovy

```diff
@@ -16,10 +16,79 @@
 package net.wasdev.wlp.gradle.plugins.tasks
 
 import net.wasdev.wlp.gradle.plugins.Liberty
+
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputFile
 import org.gradle.api.tasks.TaskAction
 
 class CreateTask extends AbstractServerTask {
 
+    String defaultPath = project.projectDir.toString() + '/src/main/liberty/config/'
+
+    CreateTask() {
+        outputs.upToDateWhen {
+            getServerDir(project).exists() && new File(getServerDir(project), 'server.xml')
+        }
+    }
+
+    @InputDirectory @Optional
+    File getConfigDir() {
+        if(server.configDirectory != null && server.configDirectory.exists()) {
+            return server.configDirectory
+        }
+    }
+
+    @InputFile @Optional
+    File getConfigFile() {
+        if(!server.configFile.toString().equals('default') && server.configFile.exists()) {
+            return server.configFile
+        } else if (server.configDirectory != null && new File(server.configDirectory, 'server.xml').exists()) {
+            return new File(server.configDirectory, 'server.xml')
+        } else if (new File(defaultPath + 'server.xml').exists()) {
+            return new File(project.projectDir.toString() + '/src/main/liberty/config/server.xml')
+        }
+    }
+
+    @InputFile @Optional
+    File getBootstrapPropertiesFile() {
+        if (!server.bootstrapPropertiesFile.toString().equals('default') && server.bootstrapPropertiesFile.exists()) {
+            return server.bootstrapPropertiesFile
+        } else if (server.configDirectory != null && new File(server.configDirectory, 'bootstrap.properties').exists()) {
+            return new File(server.configDirectory, 'bootstrap.properties')
+        } else if (new File(defaultPath + 'bootstrap.properties').exists()) {
+            return new File(defaultPath + 'bootstrap.properties')
+        }
+    }
+
+    @InputFile @Optional
+    File getJvmOptionsFile() {
+        if (!server.jvmOptionsFile.toString().equals('default') && server.jvmOptionsFile.exists()) {
+            return server.jvmOptionsFile
```

Comment:
After seeing this logic three times now, why not abstract this out into one method and just pass in the file to check and the file name?